Source/Core/Common/Subscribable.h

```diff
@@ -0,0 +1,78 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <cstdint>
+#include <functional>
+#include <map>
+#include <memory>
+
+#include "Common/NoDiscard.h"
+#include "Common/NonCopyable.h"
+
+template <typename... Args>
+class Subscribable : NonCopyable
+{
+public:
+  using SubscriptionID = uint32_t;
+  using CallbackType =
+      std::function<void(std::add_lvalue_reference_t<std::add_const_t<std::decay_t<Args>>>...)>;
+
+  class Subscription
+  {
+    friend class Subscribable;
+
+  public:
+    Subscription() = default;
```

Comment:
What's this ctor used for? Is this necessary for the members in `CFrame` and `MainWindow`?
Because it otherwise doesn't make much sense to construct an empty instance of this without a valid Subscription ID and reference to the Subscribable.